### PR TITLE
NumberBox: tabindex attribute shouldn't be removed after changing min/max/step options (T1090255)

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js [hunk 1]
```diff
@@ -1663,6 +1663,22 @@ QUnit.module('regressions', {
 
         assert.ok(!SpinButton.getInstance($spinButton).option('disabled'), 'spin button disabled state is correct');
     });
+
+    ['min', 'max', 'step'].forEach(optionName => {
+        QUnit.test(`T1090255 - changing ${optionName} option shouldn't lead to change tabIndex attribute`, function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js]: We usually name tests using pattern "... (T1090255)" because ticket ID is just a mark for history, not functionality creating reason

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js [hunk 2]
```diff
@@ -1663,6 +1663,22 @@ QUnit.module('regressions', {
...
+            const instance = $('#widget').dxNumberBox({
+                tabIndex: 3,
+                value: 1,
+                min: 1
+            }).dxNumberBox('instance');
+
+            const $input = instance.$element().find(`.${INPUT_CLASS}`);
+            assert.equal($input.attr('tabIndex'), '3', 'tabIndex is correct after initializing');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js]: this assert does not match the name of the test. I think, we can improve the test atomicity

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js [hunk 3]
```diff
@@ -1663,6 +1663,22 @@ QUnit.module('regressions', {
...
+
+            instance.option(optionName, 4);
+            assert.equal($input.attr('tabIndex'), '3', 'tabIndex is correct after option changed');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js]: we try to use `strictEqual` everywhere. You don't use `==` operator, do you?

File: js/ui/number_box/number_box.base.js [hunk 4]
```diff
@@ -232,17 +232,13 @@ const NumberBoxBase = TextEditor.inherit({
...
-            'min': this.option('min'),
-            'max': this.option('max'),
-            'step': this.option('step')
-        });
+        this._input().prop({ min, max, step });
 
         this.setAria({
-            'valuemin': ensureDefined(this.option('min'), ''),
-            'valuemax': ensureDefined(this.option('max'), '')
+            'valuemin': ensureDefined(min, ''),
```
Comment (ksercs) [inline js/ui/number_box/number_box.base.js]: we can also use new nullish coalescing operator
```suggestion
            'valuemin': min ?? '',
```

**Comment by ksercs** [comment]
@EugeniyKiyashko  we usually name pull requests describing how should it work or what we changed, the actual name seems strange to me - looks like it's expected that tabindex attribute is removed

It also looks prettier for me when PR is named starting from upper latter. It's a matter of taste but the most pull requests in out repository are named this way

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js (hunk 3 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js]: Yes, I fully agree with you.

File: js/ui/number_box/number_box.base.js (hunk 4 above)
Comment (EugeniyKiyashko) [reply js/ui/number_box/number_box.base.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js (hunk 1 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js (hunk 2 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js [hunk 5]
```diff
@@ -1663,6 +1663,32 @@ QUnit.module('regressions', {
 
         assert.ok(!SpinButton.getInstance($spinButton).option('disabled'), 'spin button disabled state is correct');
     });
+
+    QUnit.test('tabindex attribute should be exists after initialization (T1090255)', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js]:
```suggestion
    QUnit.test('tabindex attribute should be rendered after initialization (T1090255)', function(assert) {
```

Or "should exist on element".  "should be exist" is an incorrect english phrase

Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js]: this test is not related to the ticket issue => it's not necessary to add the ID

**Review by ksercs — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js (hunk 5 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js]: https://github.com/DevExpress/DevExtreme/pull/21796/commits/d8551cbe8adf8266953d361e37319d330d9c0897

Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/common.tests.js]: https://github.com/DevExpress/DevExtreme/pull/21796/commits/d8551cbe8adf8266953d361e37319d330d9c0897

**Review by ksercs — APPROVED** [review]

**Review by Zedwag — APPROVED** [review]
